encoding/unstable_hex_stream.ts

```diff
@@ -2,96 +2,151 @@
+ * import { encodeHex } from "@std/encoding/unstable-hex";
+ * import { HexEncoderStream } from "@std/encoding/unstable-hex-stream";
+ * import { toText } from "@std/streams";
  *
- * const stream = ReadableStream.from(["48656c6c6f2c", "20776f726c6421"])
- *   .pipeThrough(new HexDecoderStream())
- *   .pipeThrough(new TextDecoderStream());
+ * const readable = (await Deno.open("./deno.lock"))
+ *   .readable
+ *   .pipeThrough(new HexEncoderStream({ output: "string" }));
  *
- * assertEquals(await toText(stream), "Hello, world!");
+ * assertEquals(
+ *   await toText(readable),
+ *   encodeHex(await Deno.readFile("./deno.lock")),
+ * );
  * ```
  *
  * @experimental **UNSTABLE**: New API, yet to be vetted.
  *
  * @module
  */
 
-import { decodeHex, encodeHex } from "./hex.ts";
 import type { Uint8Array_ } from "./_types.ts";
 export type { Uint8Array_ };
+import {
+  calcMax,
+  decodeRawHex as decode,
+  encodeRawHex as encode,
+} from "./unstable_hex.ts";
+import { detach } from "./_common_detach.ts";
+
+type Expect<T> = T extends "bytes" ? Uint8Array_ : string;
 
 /**
- * Converts a Uint8Array stream into a hex-encoded stream.
+ * Transforms a {@linkcode Uint8Array<ArrayBuffer>} stream into a hexadecimal stream.
  *
  * @experimental **UNSTABLE**: New API, yet to be vetted.
  *
- * @see {@link https://www.rfc-editor.org/rfc/rfc4648.html#section-8}
+ * @typeParam T The type of the hexadecimal stream.
  *
- * @example Usage
+ * @example Basic Usage
  * ```ts
  * import { assertEquals } from "@std/assert";
- * import { encodeHex } from "@std/encoding/hex";
+ * import { encodeHex } from "@std/encoding/unstable-hex";
  * import { HexEncoderStream } from "@std/encoding/unstable-hex-stream";
- * import { toText } from "@std/streams/to-text";
+ * import { toText } from "@std/streams";
  *
- * const stream = ReadableStream.from(["Hello,", " world!"])
- *   .pipeThrough(new TextEncoderStream())
- *   .pipeThrough(new HexEncoderStream());
+ * const readable = (await Deno.open("./deno.lock"))
+ *   .readable
+ *   .pipeThrough(new HexEncoderStream({ output: "string" }));
  *
- * assertEquals(await toText(stream), encodeHex(new TextEncoder().encode("Hello, world!")));
+ * assertEquals(
+ *   await toText(readable),
+ *   encodeHex(await Deno.readFile("./deno.lock")),
+ * );
  * ```
  */
-export class HexEncoderStream extends TransformStream<Uint8Array, string> {
-  constructor() {
+export class HexEncoderStream<T extends "string" | "bytes">
+  extends TransformStream<
+    Uint8Array_,
+    T extends "bytes" ? Uint8Array_ : string
+  > {
+  /**
+   * Constructs a new instance.
+   *
+   * @param options The options for the hexadecimal stream.
+   */
+  constructor(options: { output?: T } = {}) {
+    const decode = function (): (input: Uint8Array_) => Expect<T> {
+      if (options.output === "bytes") return (x) => x as Expect<T>;
+      const decoder = new TextDecoder();
+      return (x) => decoder.decode(x) as Expect<T>;
```

Comment:
We can save some memory allocation by moving `const decoder = new TextDecoder();` outside of `decode` arrow function.

Also I think we need to pass `stream: true` option to `decoder.decode` call to allow partial data passed to it

Reply:
With moving decoder out to the top level scope, I think it would be better done in a separate pull request since this is the last one of the lot for streaming. 

With the stream option, we're only sending it acsii bytes so there won't be any partial data.